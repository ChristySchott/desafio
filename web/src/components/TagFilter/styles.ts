import styled from 'styled-components';

export const Container = styled.div`
  border: 1px solid #c61a28;
  border-radius: 4px;
  padding: 5px;
  height: 33px;
  display: flex;
  align-items: center;
  padding-left: 7px;

  @media (min-width: 768px) {
  }
`;

export const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;

  span {
    font-size: 1.4rem;
    color: #c61a28;

    @media (min-width: 768px) {
      font-size: 1.6rem;
    }
  }

  button {
    border: none;
    background: none;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-left: 8px;
    width: 20px;
    height: 20px;
  }
`;
