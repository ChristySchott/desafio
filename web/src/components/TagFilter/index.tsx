import React from 'react';

import { FiX } from 'react-icons/fi';

import { Container, Content } from './styles';

interface TagFilterProps {
  name: string;
  onRemove?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const TagFilter: React.FC<TagFilterProps> = ({ name, onRemove }) => (
  <Container>
    <Content>
      <span>{name}</span>
      <button type="button" onClick={onRemove}>
        <FiX size={18} />
      </button>
    </Content>
  </Container>
);

export default TagFilter;
