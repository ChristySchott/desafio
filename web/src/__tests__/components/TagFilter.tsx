import React from 'react';
import { render } from '@testing-library/react';

import TagFilter from '../../components/TagFilter';

describe('<TagFilter />', () => {
  it('should render with no error', () => {
    const { container } = render(<TagFilter name="" />);

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
