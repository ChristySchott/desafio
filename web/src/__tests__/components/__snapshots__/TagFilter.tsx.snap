// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagFilter /> should render with no error 1`] = `
<div
  class="sc-bdnylx kgGxPL"
>
  <div
    class="sc-gtssRu cJQNRt"
  >
    <span />
    <button
      type="button"
    >
      <svg
        fill="none"
        height="18"
        size="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </button>
  </div>
</div>
`;
